src/loader.rs

```diff
@@ -12,20 +48,25 @@ use crate::{Error, FilesystemResourceCache, Map, ResourceCache, Result, Tileset}
 /// intermediate artifacts, so using a type for creation can ensure that the cache is reused if
 /// loading more than one object is required.
 #[derive(Debug, Clone)]
-pub struct Loader<Cache: ResourceCache = FilesystemResourceCache> {
+pub struct Loader<
+    Cache: ResourceCache = FilesystemResourceCache,
+    Reader: ResourceReader = FilesystemResourceReader,
+> {
     cache: Cache,
+    reader: Reader,
 }
 
 impl Loader<FilesystemResourceCache> {
     /// Creates a new loader, creating a default ([`FilesystemResourceCache`]) resource cache in the process.
     pub fn new() -> Self {
         Self {
             cache: FilesystemResourceCache::new(),
```

Comment:
The ResourceCache trait was a temporal fix while we decided which path type to use. Now that there is only going to be `Path`, the trait is now irrelevant, and we can use a struct instead, afaik.

Reply:
Right, though its API never abstracted the `Path` type, only the way the resources are cached. I guess it could be used to integrate with an asset manager that already caches resources elsewhere, rather than using its own hash map. So maybe just renaming to `DefaultResourceCache` might be good for now, unless we can be sure that it's use-case isn't practical.

Reply:
Right, we'll keep the trait then and I'll rename the default one.